loki/transformations/inline.py

```diff
@@ -606,6 +606,18 @@ def inline_marked_subroutines(routine, allowed_aliases=None, adjust_imports=True
                 # Remove import if no further symbols used, otherwise clone with new symbols
                 import_map[impt] = impt.clone(symbols=new_symbols) if new_symbols else None
 
+        # Remove explicit interfaces of inlined routines
+        for intf in FindNodes(Interface).visit(routine.ir):
+            if not intf.spec:
+                _body = []
+                for s in intf.symbols:
+                    if s.name not in callees or s.name in not_inlined:
+                        _body += [s.type.dtype.procedure,]
```

Comment:
This could be done with a list comprehension:
```suggestion
                _body = tuple(
                    s.type.dtype.procedure for s in intf.symbols
                    if s.name not in callees or s.name in not_inlined
                )
```
Also allows to directly build a tuple, avoiding the `as_tuple` further down.